### PR TITLE
add kmnist dataset, mnist wrapper, 2d autocoder demo

### PR DESCRIPTION
**Pull request by brettkoonce**

### CODE REVIEW

File: Autoencoder/Autoencoder2D/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Simple 2D Autoencoder
+
+This is an example of a simple 2d-dimensional autoencoder model, using KuzushijiMNIST as a training dataset. It should produce output similar to the following:
```
Comment (texasmichelle) [inline Autoencoder/Autoencoder2D/README.md]: s/2d-dimensional/2-dimensional/

File: Datasets/MNIST/MNISTDatasetHandler.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Original source:
+// "KMNIST Dataset" (created by CODH), https://arxiv.org/abs/1812.01718
+// adapted from "Kuzushiji Dataset" (created by NIJL and others), doi:10.20676/00000341
```
Comment (texasmichelle) [inline Datasets/MNIST/MNISTDatasetHandler.swift]: This looks like a copy/paste artifact. Was it your intention to include in this file?

**Comment by texasmichelle** [comment]
This looks like a great addition! Autoencoder1D is running just fine, but I get this failure when running the new example:
```
$ swift run -c release Autoencoder2D
Loading resource: train-images-idx3-ubyte
Loading resource: train-labels-idx1-ubyte
Loading resource: t10k-images-idx3-ubyte
Loading resource: t10k-labels-idx1-ubyte
Fatal error: The Conv2D op currently does not support grouped convolutions on the CPU. A grouped convolution was attempted to be run because the input depth of 8 does not match the filter input depth of 4: file /swift-base/tensorflow-swift-apis/Sources/TensorFlow/Bindings/EagerExecution.swift, line 300
```

I'm running with the 0.7.0 release toolchain.

File: Autoencoder/Autoencoder2D/README.md (hunk 1 above)
Comment (brettkoonce) [reply Autoencoder/Autoencoder2D/README.md]: tweaked!

File: Datasets/MNIST/MNISTDatasetHandler.swift (hunk 2 above)
Comment (brettkoonce) [reply Datasets/MNIST/MNISTDatasetHandler.swift]: removed!

**Comment by brettkoonce** [comment]
Can reproduce the CPU crasher, will investigate.

**Comment by brettkoonce** [comment]
it be hitting this: https://github.com/tensorflow/tensorflow/blob/064e1535a7ba3cb9f67b9d9171309a1e9ebca2b0/tensorflow/core/kernels/conv_ops.cc#L161

**Comment by brettkoonce** [comment]
@texasmichelle not sure why the gpu mode let things pass, but the above should work now!

**Comment by brettkoonce** [comment]
@texasmichelle here you go!

**Comment by brettkoonce** [comment]
@BradLarson should work now!

**Comment by brettkoonce** [comment]
@BradLarson one more try...

**Comment by BradLarson** [comment]
Great, thanks for fixing all the minor issues.

**Pull request merged**